src/Kernel-Tests/SpecialSelectorTest.class.st

```diff
@@ -0,0 +1,55 @@
+"
+Tests related to special selector used bu the VM for some callback messages
+"
+Class {
+	#name : #SpecialSelectorTest,
+	#superclass : #TestCase,
+	#category : #'Kernel-Tests-Special'
+}
+
+{ #category : #tests }
+SpecialSelectorTest >> testCannotInterpret [
+	"Unfortunately, cannotInterpret: is currently broken.
+	The result of `cannotInterpret:` seems to be used as a receiver for the original message AGAIN.
+	This is likely a VM bug :("
+
+	| class object result |
+	"Create a new anonymous subclass of TestSpecialSelector with a nil methodDict"
+	class := Behavior new.
+	class basicSuperclass: TestSpecialSelector.
+	class methodDict: nil.
+	object := class new.
+	"Any message send to `object` should callback cannotInterpret:"
+	result := object max: 42.
+	self assert: result key equals: #cannotInterpret:.
+	self assert: result value selector equals: #max:.
+	self assert: result value arguments equals: #( 42 ).
+	self assert: result value lookupClass equals: TestSpecialSelector
+]
+
+{ #category : #tests }
+SpecialSelectorTest >> testCannotInterpretProtoObject [
+	"Test the implementation of ProtoObject>>cannotInterpret:
+	Is currenlty broken and goes in an infinite recursion of Context>>nextHandlerContext"
+
+	| class object |
+	"Create a new anonymous subclass of ProtoObject with a nil methodDict"
+	class := Behavior new.
+	class basicSuperclass: ProtoObject.
+	class methodDict: nil.
+	object := class new.
+	"Any message send to `object` should callback cannotInterpret: and raise an error"
+	self should: [ object max: 42 ] raise: Error
```

Comment:
Why would you get an exception here?
- object receives max:
- lookup finds nil method dictionary, lookup is restarted from `ProtoObject` looking for `cannotInterpret:`
- The selector is found (redefined below in this PR), the method in ProtoObject executed
- `ProtoObject>>cannotInterpret:` looks up `max:` but that would raise an exception because `nil` method dictionaries are not expected, and recurse (?)
- But if it would be found, it would return `self`. As soon as another message is sent to it, we enter in loop again!

Reply:
as written above, the manual look up of `max:` is ok and return nil (no error here).